examples/image-render-to-texture.rs

```diff
@@ -0,0 +1,183 @@
+struct Viewport {
+    image_handle: Option<Handle<Image>>,
+    size         :Extent3d,
+}
+
+impl Default for Viewport {
+    fn default() -> Self {
+        let size = Extent3d {
+            width: 512,
+            height: 512,
+            ..default()
+        };
+        Viewport {
+            image_handle : None,
+            size         :size,
+        }
+    }
+}
+
+fn setup_viewport(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut images: ResMut<Assets<Image>>,
+    mut viewport: ResMut<Viewport>,
+) {
+    let size = viewport.size;
+
+    // This is the texture that will be rendered to.
+    let mut image = Image {
+        texture_descriptor: TextureDescriptor {
+            label: None,
+            size,
+            dimension: TextureDimension::D2,
+            format: TextureFormat::Bgra8UnormSrgb,
+            mip_level_count: 1,
+            sample_count: 1,
+            usage: TextureUsages::TEXTURE_BINDING
+                | TextureUsages::COPY_DST
+                | TextureUsages::RENDER_ATTACHMENT,
+        },
+        ..default()
+    };
+
+
+    // fill image.data with zeroes
+    image.resize(size);
+
+    let image_handle = images.add(image);
+    viewport.image_handle = Some(image_handle.clone());
+
+    let cube_handle = meshes.add(Mesh::from(shape::Cube { size: 4.0 }));
+    let cube_material_handle = materials.add(StandardMaterial {
+        base_color: Color::rgb(0.8, 0.7, 0.6),
+        reflectance: 0.02,
+        unlit: false,
+        ..default()
+    });
+
+    // This specifies the layer used for the first pass, which will be attached to the first pass camera and cube.
+    let viewport_pass_layer = RenderLayers::layer(1);
+
+    // The cube that will be rendered to the texture.
+    commands.spawn((
+        PbrBundle {
+            mesh: cube_handle,
+            material: cube_material_handle,
+            transform: Transform::from_translation(Vec3::new(0.0, 0.0, 1.0)),
+            ..default()
+        },
+        FirstPassViewport,
+        viewport_pass_layer,
+    ));
+
+    // Light
+    // NOTE: Currently lights are shared between passes - see https://github.com/bevyengine/bevy/issues/3462
+    commands.spawn(PointLightBundle {
+        transform: Transform::from_translation(Vec3::new(0.0, 0.0, 10.0)),
+        ..default()
+    });
+
+    commands
+    .spawn(Camera3dBundle {
+        camera_3d: Camera3d {
+            clear_color: ClearColorConfig::Custom(Color::WHITE),
+            ..default()
+        },
+        camera: Camera {
+            // render before the "main pass" camera
+            priority: -1,
+            target: RenderTarget::Image(image_handle.clone()),
+            ..default()
+        },
+        transform: Transform::from_translation(Vec3::new(0.0, 0.0, 15.0))
+            .looking_at(Vec3::default(), Vec3::Y),
+        ..default()
+    })
+    .insert(viewport_pass_layer);

```

Comment:
I found that the render UICamera doesn't respects the RenderLayer: https://github.com/bevyengine/bevy/issues/6069 and the resulting ui renders with artifacts (if I change background-color for image, for example).

Provided in the issue workaround works like a charm, so my suggestion is to notice the bug in example and add
```rust
.insert(UiCameraConfig { show_ui: false })
```